pkg/sdn/plugin/subnets.go

```diff
@@ -48,33 +48,38 @@ func (master *OsdnMaster) SubnetStartMaster(clusterNetwork *net.IPNet, hostSubne
 	return nil
 }
 
-func (master *OsdnMaster) addNode(nodeName string, nodeIP string, hsAnnotations map[string]string) error {
+// addNode takes the nodeName, a preferred nodeIP, the node's annotations and other valid ip addresses
+// Creates or updates a HostSubnet if needed
+// Returns the IP address used for hostsubnet (either the preferred or one from the otherValidAddresses) and any error
+func (master *OsdnMaster) addNode(nodeName string, nodeIP string, hsAnnotations map[string]string, otherValidAddresses []kapi.NodeAddress) (string, error) {
```

Comment:
The F5 ghost node thing prevented me from doing that.. F5 exists as a hostsubnet, but there is no Node object for it.